students/brandon_nguyen/session09/mailroom_oo/donor_model.py

```diff
@@ -51,15 +68,47 @@ def __str__(self):
 
 
 class DonorCollection:
-    # Revew mailroom 1 instruction to do the following
+    """ This class will hold:
+            -all Donor objects
+            -methods to add new donor, search for a given donor
+            - extract donor
+            -create report
+            -send thank you letters to all
+    """
+    def __init__(self):
+        self.__donor_db = {}
     # We need data structure and other constants for printing
 
-    # method to update donation
-
-    # method to return donor in list getName?
+    @property
+    def donor_db(self):
+        return self.__donor_db
 
-    # method create report
+    # method to update donation
+    def add_donor_db(self, name, donation=None):
+        """ To add donor into collection"""
+        # How best to handle existing donor?
+        # TODO existing donor only take donation?
+
+        # assume donation is ok TODO more
+        self.__donor_db[name] = Donor(name)
+        self.__donor_db[name].add_donation(donation)
+
+    def list_donors(self):
+        """ quick way to return a list of names in db """
+        lst_names = []
+        [lst_names.append(name) for name in self.__donor_db.keys()]
```

Comment:
Again, this code pattern works, but not a desired one (showing lack of full understanding of the list comprehension). Please change the two lines above to this:
```
lst_names = [name for name in self.__donor_db.keys()]
```
However, the list comprehension is not even needed at all in this case. You can simply say as follows, to get a list of all keys in a dict:
```
lst_names = list(self.__donor_db.keys())
```
This would be considered better than the ones above.

Reply:
Thank you!  This is much, much cleaner!